spyder/widgets/findreplace.py

```diff
@@ -571,12 +571,20 @@ def replace_find_selection(self, focus_replace_text=False):
             replacement = re_pattern.sub(replace_text, selected_text)
             if replacement != selected_text:
                 cursor = self.editor.textCursor()
+                start_pos = cursor.selectionStart()
                 cursor.beginEditBlock()
                 cursor.removeSelectedText()
                 if not self.re_button.isChecked():
                     replacement = re.sub(r'\\(?![nrtf])(.)', r'\1', replacement)
                 cursor.insertText(replacement)
+                # Restore selection
+                self.editor.set_cursor_position(start_pos)
+                sel_len = len(replacement) - \
+                            replacement.count(self.editor.get_line_separator())

```

Comment:
@oscargus thanks for the change!

We generally discourage the use of the `\` for continuation lines, so maybe if it is still too long we could do something like

```python
                # Restore selection
                self.editor.set_cursor_position(start_pos)
                separator = self.editor.get_line_separator()
                selection_len = len(replacement) - replacement.count(separator)
```